From f1299f9eb199d51af3b95f3bf7d19f66600f8c0c Mon Sep 17 00:00:00 2001
From: Ajit Vaishya <ajitv@codeaurora.org>
Date: Wed, 23 May 2018 22:31:29 +0530
Subject: [PATCH 16/18] sepolicy: Allow android service to write on tombstones

With this commit added sepolicy rule, so that
android.hardware.wifi@1.0-service support to write
on tombstones path.
i.e /data/vendor/tombstones/wifi

Change-Id: I53633c5291f27041f23726e3d8426eab1adcd664
CRs-Fixed: 2241490

# Conflicts:
#	common/hal_wifi.te
---
 common/hal_wifi.te | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/common/hal_wifi.te b/common/hal_wifi.te
index 0974817..aa0bfe3 100644
--- a/common/hal_wifi.te
+++ b/common/hal_wifi.te
@@ -36,5 +36,9 @@ allow hal_wifi proc_modules:file r_file_perms;
 allow hal_wifi_default location_data_file:dir search;
 allow hal_wifi_default location:unix_stream_socket connectto;
 allow hal_wifi_default location_socket:sock_file write;
-# /data/vendor/tombstones/wifi
-r_dir_file(hal_wifi_default, vendor_tombstone_data_file);
+
+# allow hal_wifi to write into /data/vendor/tombstones/wifi
+userdebug_or_eng(`
+  allow hal_wifi_server vendor_tombstone_data_file:dir rw_dir_perms;
+  allow hal_wifi_server vendor_tombstone_data_file:file create_file_perms;
+')
-- 
2.17.1

